% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{book}
\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{booktabs}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage[]{natbib}
\bibliographystyle{plainnat}
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={NCGR Bioinformatics Workshops},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{NCGR Bioinformatics Workshops}
\author{}
\date{\vspace{-2.5em}}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{1}
\tableofcontents
}
\hypertarget{license-and-copyright}{%
\chapter*{License and Copyright}\label{license-and-copyright}}
\addcontentsline{toc}{chapter}{License and Copyright}

Creative Commons Attribution-NonCommercial-NoDerivatives 4.0
\url{https://creativecommons.org/licenses/by-nc-nd/4.0/}

Â© 2023 National Center for Genome Resources

\includegraphics[width=0.2\textwidth,height=\textheight]{./Figures/INBRE_Logo_Grad_transparent-2019.png}

\includegraphics[width=0.3\textwidth,height=\textheight]{./Figures/ncgr.png}

\hypertarget{upcoming-ncgr-workshops}{%
\chapter*{Upcoming NCGR Workshops}\label{upcoming-ncgr-workshops}}
\addcontentsline{toc}{chapter}{Upcoming NCGR Workshops}

\hypertarget{pangenomics-workshop-some-bioinformatics-experience}{%
\subsection*{Pangenomics Workshop (some bioinformatics experience)}\label{pangenomics-workshop-some-bioinformatics-experience}}
\addcontentsline{toc}{subsection}{Pangenomics Workshop (some bioinformatics experience)}

\begin{itemize}
\item
  Fill out this online application here:
  \href{https://forms.gle/49UbSZFk6AxcGWcH8}{\textbf{Pangenomics Workshop Application}}
\item
  Application \textbf{due} by midnight \textbf{February 17, 2025}
\item
  Space is limited!
\end{itemize}

\hypertarget{when}{%
\subsubsection*{When}\label{when}}
\addcontentsline{toc}{subsubsection}{When}

\textbf{March 03-07, 2025} (5 days)

9 am -- 4 pm

\begin{itemize}
\tightlist
\item
  More Info about \href{https://inbre.ncgr.org/ncgr-workshops/pangenomics-workshop.html\#pangenomics-workshop}{\textbf{Pangenomics Workshops here}}
\end{itemize}

\hypertarget{how-much-does-it-cost}{%
\subsubsection*{How much does it cost?}\label{how-much-does-it-cost}}
\addcontentsline{toc}{subsubsection}{How much does it cost?}

\href{https://inbre.ncgr.org/ncgr-workshops/workshop-rates.html}{\textbf{Click here}} to get workshop rates.

\hypertarget{single-cell-workshop}{%
\subsection*{Single Cell Workshop}\label{single-cell-workshop}}
\addcontentsline{toc}{subsection}{Single Cell Workshop}

\begin{itemize}
\item
  Fill out this online application here:
  \href{https://forms.gle/xcfJYaHSDr4SQYUw9}{\textbf{Single Cell Workshop Application}}
\item
  Application \textbf{due} by midnight \textbf{March 03, 2025}
\item
  Space is limited!
\end{itemize}

\hypertarget{when-1}{%
\subsubsection*{When}\label{when-1}}
\addcontentsline{toc}{subsubsection}{When}

\textbf{March 17-21, 2025} (5 days)

9 am -- 4 pm

\begin{itemize}
\tightlist
\item
  More Info about \href{https://inbre.ncgr.org/ncgr-workshops/single-cell-rna-seq-workshop.html\#single-cell-rna-seq-workshop}{\textbf{Single Cell Workshops here}}
\end{itemize}

\hypertarget{how-much-does-it-cost-1}{%
\subsubsection*{How much does it cost?}\label{how-much-does-it-cost-1}}
\addcontentsline{toc}{subsubsection}{How much does it cost?}

\href{https://inbre.ncgr.org/ncgr-workshops/workshop-rates.html}{\textbf{Click here}} to get workshop rates.

\includegraphics[width=0.5\textwidth,height=\textheight]{./Figures/ncgr.png}

\hypertarget{contact}{%
\subsubsection*{Contact}\label{contact}}
\addcontentsline{toc}{subsubsection}{Contact}

Please contact Project Coordinator, Ethan Price with any additional questions you may have at \href{mailto:inbre@ncgr.org}{\nolinkurl{inbre@ncgr.org}} and we look forward to your participation!

Sponsored by New Mexico IDeA Network for Biomedical Research Excellence (NM-INBRE).

\hypertarget{workshop-calendar}{%
\chapter*{Workshop Calendar}\label{workshop-calendar}}
\addcontentsline{toc}{chapter}{Workshop Calendar}

\hypertarget{march-03-07-2025-pangenomics-workshop}{%
\subsubsection*{March 03-07, 2025 \textbar{} Pangenomics Workshop}\label{march-03-07-2025-pangenomics-workshop}}
\addcontentsline{toc}{subsubsection}{March 03-07, 2025 \textbar{} Pangenomics Workshop}

\hypertarget{march-17-21-2025-single-cell-workshop}{%
\subsubsection*{March 17-21, 2025 \textbar{} Single Cell Workshop}\label{march-17-21-2025-single-cell-workshop}}
\addcontentsline{toc}{subsubsection}{March 17-21, 2025 \textbar{} Single Cell Workshop}

\hypertarget{november-18th---november-22nd-2024-differential-expression-workshop}{%
\subsubsection*{November 18th - November 22nd, 2024 \textbar{} Differential Expression Workshop!}\label{november-18th---november-22nd-2024-differential-expression-workshop}}
\addcontentsline{toc}{subsubsection}{November 18th - November 22nd, 2024 \textbar{} Differential Expression Workshop!}

\hypertarget{november-4th---november-8th-2024-metagenomics-workshop}{%
\subsubsection*{November 4th - November 8th, 2024 \textbar{} Metagenomics Workshop!}\label{november-4th---november-8th-2024-metagenomics-workshop}}
\addcontentsline{toc}{subsubsection}{November 4th - November 8th, 2024 \textbar{} Metagenomics Workshop!}

\textless{}\textgreater{}

\hypertarget{about-us}{%
\chapter*{About Us}\label{about-us}}
\addcontentsline{toc}{chapter}{About Us}

\textbf{NM-INBRE (New Mexico IDeA Networks of Biomedical Research Excellence):}

\includegraphics[width=0.2\textwidth,height=\textheight]{./Figures/INBRE_Logo_Grad_transparent-2019.png}

NM-INBRE Champions biomedical and community based research excellence in the state of New Mexico through the development of innovative, supportive and sustainable research environments for faculty and students, community engaging health initiatives, while building a network of lead scientists and educators at the state, regional and national level.

Funded by the Institutional Development Award (\textbf{IDeA}) of the National Institutes of Health (\textbf{NIH}) \& National Institute of General Medical Sciences (\textbf{NIGMS}) Grant \# P20GM103451

\textbf{NCGR (National Center for Genome Research):}

\includegraphics[width=0.3\textwidth,height=\textheight]{./Figures/ncgr.png}

Located in Santa Fe, New Mexico, the National Center for Genome Resources is a not-for-profit research institute that innovates, collaborates, and educates in the field of genomic data science. As leaders in DNA sequence analysis, we partner with government, industry, and academia to drive biological discovery in all kingdoms of life.

NCGR operates NM-INBRE's Sequencing and Bioinformatics Core (SBC). Through this partnership we offer the NM-INBRE SBC Pilot Award, which is a funding opportunity primarily for New Mexico researchers through which NCGR can help you with your research.

\hypertarget{workshop-rates}{%
\chapter*{Workshop Rates}\label{workshop-rates}}
\addcontentsline{toc}{chapter}{Workshop Rates}

\includegraphics[width=0.9\textwidth,height=\textheight]{./Figures/WorkshopRates.png}

\hypertarget{new-mexico-scholarship-link}{%
\subsubsection*{* New Mexico Scholarship Link:}\label{new-mexico-scholarship-link}}
\addcontentsline{toc}{subsubsection}{* New Mexico Scholarship Link:}

\href{https://forms.gle/MkgHEQGCBMkcVYBX8}{\textbf{Scholarship link - Click here}} to apply for a \textbf{New Mexico} scholarship!

\begin{itemize}
\tightlist
\item
  There are a limited number of scholarships available per workshop.
\item
  Scholarships only available to New Mexico students and researchers.
\item
  We will contact workshop and scholarship applicants within a week of the application deadline.
\item
  Check with \textbf{your institution's INBRE program} for other funding opportunities!
\end{itemize}

For more information about \href{https://www.nigms.nih.gov/capacity-building/division-for-research-capacity-building/institutional-development-award-\%28idea\%29}{Institutional Development Award (IDeA)}

\hypertarget{pangenomics-workshop}{%
\chapter*{Pangenomics Workshop}\label{pangenomics-workshop}}
\addcontentsline{toc}{chapter}{Pangenomics Workshop}

Application: Fill out this online application here: \href{https://docs.google.com/forms/d/e/1FAIpQLScJPueRpT0M4CHb2RjJI5DGHTXu6mutGBVBoOMCEQsO2JnQ1A/viewform}{Pangenomics Workshop Application}

Space is limited!

\hypertarget{when-2}{%
\section*{When}\label{when-2}}
\addcontentsline{toc}{section}{When}

March 03-07, 2025

\hypertarget{where}{%
\section*{Where}\label{where}}
\addcontentsline{toc}{section}{Where}

Virtual (by the National Center for Genome Resources (NCGR) Santa Fe, NM)

\hypertarget{audience}{%
\section*{Audience}\label{audience}}
\addcontentsline{toc}{section}{Audience}

This program is geared towards training biologists (undergraduates, graduates and researchers) with some bioinformatics experience.

\hypertarget{topic-overview}{%
\section*{Topic Overview}\label{topic-overview}}
\addcontentsline{toc}{section}{Topic Overview}

The proliferation of reference quality genome assemblies within any single species has necessitated the need for pangenome analyses. Such analyses remove reference-bias and elucidate biological signals at a more comprehensive population scale. In this workshop, students will learn what exactly a pangenome is, how to build a pangenome, and how to perform fundamental bioinformatic analyses on pangenomic data.

\hypertarget{objective}{%
\section*{Objective}\label{objective}}
\addcontentsline{toc}{section}{Objective}

Students will learn to use the command-line interface to conduct graphical pangenome analyses. Specifically, students will learn to build pangenome graphs using vg, minigraph, and PGGB. They will learn to visualize these pangenomes using Bandage and IGV, and they will learn how to map reads and call variants on the graphs, comparing both run-time and mapping performance between methods. Additionally, students will be given an overview of other, interoperable tools to broaden their understanding of the rapidly moving pangenomics space.

\hypertarget{prerequisites}{%
\section*{Prerequisites}\label{prerequisites}}
\addcontentsline{toc}{section}{Prerequisites}

Reference material will be provided upon acceptance.

\hypertarget{what-to-expect-upon-acceptance}{%
\section*{What to expect upon acceptance}\label{what-to-expect-upon-acceptance}}
\addcontentsline{toc}{section}{What to expect upon acceptance}

Upon acceptance you will receive an email confirming acceptance into the internship program and further details.

\hypertarget{logistics}{%
\section*{Logistics}\label{logistics}}
\addcontentsline{toc}{section}{Logistics}

The workshop will be conducted virtually using primarily Zoom as our conferencing platform. We will also use a variety of other software and tools, and you will log onto NCGR's Unix analysis server for the command line tools. After the workshop, students can use our analysis server for their research or education purposes for up to one year. Late applications will not be accepted. A Selection Committee will select participants shortly after the deadline. If any questions arise while reviewing your application packet, we will contact you directly.

Contact

Please contact Project Coordinator, Ethan Price, with any additional questions you may have at \href{mailto:eprice@ncgr.org}{\nolinkurl{eprice@ncgr.org}}. We look forward to your participation!

Sponsored by New Mexico IDeA Network for Biomedical Research Excellence (NM-INBRE).

\hypertarget{single-cell-rna-seq-workshop}{%
\chapter*{Single Cell RNA-seq Workshop}\label{single-cell-rna-seq-workshop}}
\addcontentsline{toc}{chapter}{Single Cell RNA-seq Workshop}

Application: Fill out this online application here: \href{https://forms.gle/xcfJYaHSDr4SQYUw9}{Single Cell Workshop Application}

Space is limited!

Application due by midnight March 03, 2025

\hypertarget{when-3}{%
\section*{When}\label{when-3}}
\addcontentsline{toc}{section}{When}

March 17 -- 21, 2025 (5 days)

9 am -- 4 pm

\hypertarget{where-1}{%
\section*{Where}\label{where-1}}
\addcontentsline{toc}{section}{Where}

Virtual (by the National Center for Genome Resources (NCGR) Santa Fe, NM)

\hypertarget{objective-1}{%
\section*{Objective}\label{objective-1}}
\addcontentsline{toc}{section}{Objective}

Students will leverage their Linux and R command knowledge to become familiar with the SingleCellExperiment, Seurat R packages, Loupe Browser, and an interactive Shiny web application to perform statistical metrics and data visualization for a variety of bioinformatic approaches to single cell sequencing analysis.

\hypertarget{audience-1}{%
\section*{Audience}\label{audience-1}}
\addcontentsline{toc}{section}{Audience}

This advanced workshop is targeted towards undergraduate or graduate students in biology or related fields.

\hypertarget{prerequisites-1}{%
\section*{Prerequisites}\label{prerequisites-1}}
\addcontentsline{toc}{section}{Prerequisites}

Reference material will be provided upon acceptance.

\hypertarget{logistics-1}{%
\section*{Logistics}\label{logistics-1}}
\addcontentsline{toc}{section}{Logistics}

The workshop will be conducted virtually using primarily Zoom as our conferencing platform. We will also use a variety of other software and tools, and you will log onto NCGR's analysis server for the command line tools. After the workshop, students can use our analysis server for their research or education purposes for up to one year. Late applications will not be accepted. Participants will be accepted shortly after the deadline. If any questions arise while reviewing your application packet, we will contact you directly.

Contact

Please contact Project Coordinator, Ethan Price with any additional questions you may have at \href{mailto:inbre@ncgr.org}{\nolinkurl{inbre@ncgr.org}} and we look forward to your participation!

Sponsored by New Mexico IDeA Network for Biomedical Research Excellence (NM-INBRE).

\hypertarget{metagenomics-workshop}{%
\chapter*{Metagenomics Workshop}\label{metagenomics-workshop}}
\addcontentsline{toc}{chapter}{Metagenomics Workshop}

Application: Fill out this online application here: \href{https://forms.gle/Ai2psNtZJRvxygZZ9}{Metagenomics Workshop Application}

Space is limited!

Application due by midnight October 28th, 2024

\hypertarget{when-4}{%
\section*{When}\label{when-4}}
\addcontentsline{toc}{section}{When}

\textbf{November 04 -- November 08, 2024} (5 days)

9 am -- 4 pm

\begin{itemize}
\tightlist
\item
  Including a 1 hour break for lunch
\end{itemize}

\hypertarget{where-2}{%
\section*{Where}\label{where-2}}
\addcontentsline{toc}{section}{Where}

Virtual (via the National Center for Genome Resources (NCGR) Santa Fe, NM)

\hypertarget{objective-2}{%
\section*{Objective}\label{objective-2}}
\addcontentsline{toc}{section}{Objective}

Learn how to use the UNIX command line, analytical workflows and public tools to independently analyze sequencing data plus how to visualize and render data using graphing tools.

This workshop covers analyses of 16s and whole genome shotgun metagenomic sequence data, including community analysis and assembly.

\hypertarget{audience-2}{%
\section*{Audience}\label{audience-2}}
\addcontentsline{toc}{section}{Audience}

This workshop is targeted towards undergraduate or graduate students, and researchers in biology or related fields.

\hypertarget{prerequisites-2}{%
\section*{Prerequisites}\label{prerequisites-2}}
\addcontentsline{toc}{section}{Prerequisites}

Unix, sequencing basics and other study guides will be provided upon acceptance.

What to expect upon acceptance

Email confirming acceptance into the internship program and further details.

\hypertarget{logistics-2}{%
\section*{Logistics}\label{logistics-2}}
\addcontentsline{toc}{section}{Logistics}

The workshop will be conducted virtually using primarily Zoom as our conferencing platform. We will also use a variety of other software and tools, and you will log onto NCGR's Unix analysis server for the command line tools. After the workshop, students can use our analysis server for their research or education purposes for up to one year.

Late applications will not be accepted. Selection Committee to select participants shortly after the deadline. If any questions arise while reviewing your application packet, we will contact you directly.
Contact

Please contact Project Coordinator, Ethan Price, with any additional questions you may have at \href{mailto:inbre@ncgr.org}{\nolinkurl{inbre@ncgr.org}} and we look forward to your participation!

Sponsored by New Mexico IDeA Network for Biomedical Research Excellence (NM-INBRE).

\hypertarget{differential-expression-workshop}{%
\chapter*{Differential Expression Workshop}\label{differential-expression-workshop}}
\addcontentsline{toc}{chapter}{Differential Expression Workshop}

Application: Fill out this online application here:
\href{https://forms.gle/aXG2StdRnSfKAcfA6}{Differential Expression Workshop Application}

Space is limited!

Application due by midnight November 11th, 2024

\hypertarget{when-5}{%
\section*{When}\label{when-5}}
\addcontentsline{toc}{section}{When}

November 18 -- 22, 2024 (5 days)

9 am -- 4 pm

\hypertarget{where-3}{%
\section*{Where}\label{where-3}}
\addcontentsline{toc}{section}{Where}

Virtual (by the National Center for Genome Resources (NCGR) Santa Fe, NM)

\hypertarget{objective-3}{%
\section*{Objective}\label{objective-3}}
\addcontentsline{toc}{section}{Objective}

Come learn about differential gene expression analysis in this virtual workshop that will teach you with hands-on analysis from start to finish. This differential expression workshop covers basic linux skills, quality control, read alignment, abundance estimation, differential expression analysis, visualization, and pathway analysis. Note that this workshop covers differential expression of bulk tissues (we cover single cell expression in a separate workshop).

\hypertarget{audience-3}{%
\section*{Audience}\label{audience-3}}
\addcontentsline{toc}{section}{Audience}

This advanced workshop is targeted towards undergraduate or graduate students in biology or related fields.

\hypertarget{prerequisites-3}{%
\section*{Prerequisites}\label{prerequisites-3}}
\addcontentsline{toc}{section}{Prerequisites}

Unix, sequencing basics and other study guides will be provided upon acceptance.

What to expect upon acceptance

Email confirming acceptance into the internship program and further details.

\hypertarget{logistics-3}{%
\section*{Logistics}\label{logistics-3}}
\addcontentsline{toc}{section}{Logistics}

The workshop will be conducted virtually using primarily Zoom as our conferencing platform. We will also use a variety of other software and tools, and you will log onto NCGR's Unix analysis server for the command line tools. After the workshop, students can use our analysis server for their research or education purposes for up to one year.

Selection Committee to select participants shortly after the deadline. If any questions arise while reviewing your application packet, we will contact you directly.

Contact

Please contact Project Coordinator, Ethan Price with any additional questions you may have at \href{mailto:inbre@ncgr.org}{\nolinkurl{inbre@ncgr.org}} and we look forward to your participation!

Sponsored by New Mexico IDeA Network for Biomedical Research Excellence (NM-INBRE).

\hypertarget{citing-nm-inbre}{%
\chapter*{Citing NM INBRE}\label{citing-nm-inbre}}
\addcontentsline{toc}{chapter}{Citing NM INBRE}

If you received funding from NM-INBRE that contributed to your research or career development, you must cite NM-INBRE support on all publications, presentations, press releases, requests, requests for proposals, bid invitations, or any other documents or applications related to your funded research. Also, please be sure to cite NM-INBRE if your work benefited from the use of NM-INBRE equipment or an NM-INBRE sponsored student worked in your lab.

For instructions on how to cite NM-INBRE and where to find our logos, please visit \url{https://nminbre.nmsu.edu/cite-us/cite-us.html}

{\textbf{NIH requires the following format for citing INBRE support, to be used for presentations, publications, and other acknowledgements:}}

{\emph{Research reported in this publication was supported by an Institutional Development Award (IDeA) from the National Institute of General Medical Sciences of the National Institutes of Health under grant number P20GM103451.}}

\hypertarget{server-access-acknowledgement}{%
\chapter*{Server Access \& Acknowledgement}\label{server-access-acknowledgement}}
\addcontentsline{toc}{chapter}{Server Access \& Acknowledgement}

For those of you who requested continued access to our server, we will extend your account for 1 year. If you need it longer, please email us and we'll be happy to work with you. We ask that you don't run really large jobs while we are running workshops to avoid slowing things down. So, if you have a really large job to run, please check in with us so we can let you know when workshops will be happening.

Please note, if you use our servers to do analysis that you publish, you need to acknowledge the National Center for Genome Resources (NCGR) and the INBRE grant in your acknowledgements. Instructions for the latter can be found here: \url{https://nminbre.nmsu.edu/cite-us/cite-us.html}.

This is really important for our annual reports so thank you in advance!

Additionally, in order to share your publications in our annual reporting, and to comply with NIH guidelines for open access, please include a PMCID with your publication. Instructions: \href{https://publicaccess.nih.gov/include-pmcid-citations.htm}{Include PMCID in Citations}.

Additional details from \url{https://nminbre.nmsu.edu/cite-us/cite-us.html}:

{If you received funding from NM-INBRE that contributed to your research or career development, you must cite NM-INBRE support on all publications, presentations, press releases, requests, requests for proposals, bid invitations, or any other documents or applications related to your funded research. Also, please be sure to cite NM-INBRE if your work benefited from the use of NM-INBRE equipment or an NM-INBRE sponsored student worked in your lab.}

{\textbf{NIH requires the following format for citing INBRE support, to be used for presentations, publications, and other acknowledgements:}}

{\emph{Research reported in this publication was supported by an Institutional Development Award (IDeA) from the National Institute of General Medical Sciences of the National Institutes of Health under grant number P20GM103451.}}

\hypertarget{questions}{%
\chapter*{Questions}\label{questions}}
\addcontentsline{toc}{chapter}{Questions}

If you have \textbf{bioinformatics or technical questions}, please email \href{mailto:inbre@ncgr.org}{\nolinkurl{inbre@ncgr.org}}, which will send emails to everyone on the inbre team.

If you have questions on \textbf{passwords/password resets, payments, receipts, etc,} please email Ethan Price at \href{mailto:inbre@ncgr.org}{\nolinkurl{inbre@ncgr.org}} and cc \href{mailto:inbre@ncgr.org}{\nolinkurl{inbre@ncgr.org}}.

Occasionally, we change servers or change how to log in so if you are having issues, please contact us.

\hypertarget{acknowledgements}{%
\chapter*{Acknowledgements}\label{acknowledgements}}
\addcontentsline{toc}{chapter}{Acknowledgements}

This publication was supported by an Institutional Development Award (\textbf{IDeA}) from the National Institute of General Medical Sciences of the \textbf{National Institutes of Health} under grant number \textbf{P20GM103451}. Additional support came from \textbf{National Science Foundation} Award numbers \textbf{1759522} (Collaborative Research: Innovation: Pioneering New Approaches to Explore Pangenomic Space at Scale) and \textbf{2105391} (CRII: III: Toward the Compression of Pangenomic DNA Sequence Data Using Context-Free Grammars).

\includegraphics[width=0.4\textwidth,height=\textheight]{./Figures/INBRE_Logo_Grad_transparent-2019.png}

\includegraphics[width=0.5\textwidth,height=\textheight]{./Figures/ncgr.png}

  \bibliography{book.bib,packages.bib}

\end{document}
